src/axom/core/Array.hpp

```diff
@@ -933,6 +956,20 @@ Array<T, DIM, SPACE>::Array(const axom::StackArray<axom::IndexType, DIM>& shape,
              false);
 }
 
+//------------------------------------------------------------------------------
+template <typename T, int DIM, MemorySpace SPACE>
+Array<T, DIM, SPACE>::Array(const axom::StackArray<axom::IndexType, DIM>& shape,
+                            const axom::ArrayIndexer<DIM>& indexer,
+                            int allocator_id)
+  : ArrayBase<T, DIM, Array<T, DIM, SPACE>>(shape, indexer)
+  , m_allocator_id(allocator_id)
+{
+  assert(indexer.fastestStrideLength() == 1);
```

Comment:
Is `Array` always expected to be initialized with a min-stride of 1? If so, you might want to document that requirement for the "arbitrary-strides" constructor.

Reply:
Yes, `Array` always has min-stride of 1.  Using `ArrayIndexer` or `strides` in constructors does let users specify inconsistent and non-sensical values.  We'd have to catch these errors with asserts.  Would it make sense to instead use constructors that make this kind of error impossible?  This means allowing users to specify only `ArrayStrideOrder`, like @publixsubfan suggested above, or `axom::StackArray<DirType, DIM>& slowestDirs` parameters, to handle all possible arbitrary cases.
```
Array(const axom::StackArray<axom::IndexType, DIM>& shape,
      int allocator_id = axom::detail::getAllocatorID<SPACE>(),
      ArrayStrideOrder strideOrder = ArrayStrideOrder::ROW);

template <typename DirType>
Array(const axom::StackArray<axom::IndexType, DIM>& shape,
      int allocator_id,
      const axom::StackArray<DirType, DIM>& slowestDirs);
```

Reply:
Done.